go panic if you execute cnspec scan

if you run a cnspec scan with a policy. you can use different policies too. you get a go panic

```
cnspec scan local --policy-bundle stable/mondoo-platform-overview.mql.yaml    x1 14h2m main[5c691a4]
→ loaded configuration from /home/user/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
→ connecting to asset atomic111 (baremetal)

███████████████████████████████████████████████████████████████████████████ 100% atomic111


Asset: atomic111
================
Data queries:
os.hostname: "atomic111"
platform.arch: "x86_64"
kernel.installed: []
packages.list.length: 4068
service: {
  description: "Mondoo Service"
  enabled: true
  type: "systemd"
  running: true
  name: "mondoo"
  installed: true
  masked: false
}
platform.release: "rolling"
platform.title: "Arch Linux, bare metal"
mondoo.version: "unstable"
command.stderr: "/sys/firmware/dmi/tables/smbios_entry_point: Permission denied
/dev/mem: Permission denied
"
if: {
  command.stdout: "09/29/2022
1.57
LENOVO
N32ET81W (1.57 )
Not Available
20XWCTO1WW
... 16 more lines ...
os.uptime: 1 days 17 hours 54 minutes 3 seconds
mondoo.jobEnvironment: {
  id: "client.mondoo.com"
  name: "Mondoo Client"
}

Controls:


Scanned 1 assets

Arch Linux
    U atomic111
panic: runtime error: index out of range [1] with length 0

goroutine 1 [running]:
go.mondoo.com/cnspec/cli/reporter.(*defaultReporter).printSummary(0xc00220fb50, {0xc0001cc1a0, 0x1, 0x1010000000008?})
        /home/user/go/src/go.mondoo.io/cnspec/cli/reporter/print_compact.go:150 +0xc76
go.mondoo.com/cnspec/cli/reporter.(*defaultReporter).print(0xc00220fb50)
        /home/user/go/src/go.mondoo.io/cnspec/cli/reporter/print_compact.go:70 +0x292
go.mondoo.com/cnspec/cli/reporter.(*Reporter).Print(0x84f1370?, 0x84ea101?, {0x9567120?, 0xc0001a8008?})
        /home/user/go/src/go.mondoo.io/cnspec/cli/reporter/reporter.go:56 +0x146
go.mondoo.com/cnspec/apps/cnspec/cmd.printReports(0xc00113f200?, 0xc00113f200, 0x0?)
        /home/user/go/src/go.mondoo.io/cnspec/apps/cnspec/cmd/scan.go:581 +0x11f
go.mondoo.com/cnspec/apps/cnspec/cmd.glob..func19(0x0?, {0xc000710440?, 0x0?, 0x0?}, 0x0?, 0x0?)
        /home/user/go/src/go.mondoo.io/cnspec/apps/cnspec/cmd/scan.go:362 +0x139
go.mondoo.com/cnquery/apps/cnquery/cmd/builder.localProviderCmd.func1(0xc00156b200?, {0xc000710440?, 0x2?, 0x2?})
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230103175939-8265160317db/apps/cnquery/cmd/builder/builder.go:195 +0x2b
github.com/spf13/cobra.(*Command).execute(0xc00156b200, {0xc000710420, 0x2, 0x2})
        /home/user/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xd1cf020)
        /home/user/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/user/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
go.mondoo.com/cnspec/apps/cnspec/cmd.Execute()
        /home/user/go/src/go.mondoo.io/cnspec/apps/cnspec/cmd/root.go:76 +0x25
main.main()
        /home/user/go/src/go.mondoo.io/cnspec/apps/cnspec/cnspec.go:6 +0x17
```

--- a/cli/reporter/print_compact.go
+++ b/cli/reporter/print_compact.go
@@ -142,20 +142,20 @@ func (r *defaultReporter) printSummary(orderedAssets []assetMrnName) {
 		}
 	}
 
-	// we do not have a space url, so we extract it form the asset url
-	// https://console.mondoo.com/space/fleet/2JtqGyVTZULTW0uwQ5YxXW4nh6Y?spaceId=dazzling-golick-767384
-	// an individual asset url wouldn't make sense here
-	spaceUrlRegexp := regexp.MustCompile(`^(http.*)/[a-zA-Z0-9-]+(\?.+)$`)
-	m := spaceUrlRegexp.FindStringSubmatch(assetUrl)
-	spaceUrl := m[1] + m[2]
-
 	if len(orderedAssets) > 1 && strings.Contains(r.data.Assets[orderedAssets[0].Mrn].PlatformName, "Kubernetes") {
 		r.out.Write([]byte(NewLineCharacter))
 		r.out.Write([]byte("To scan an individual asset run `mondoo scan k8s --resource KIND:NAMESPACE:NAME`" + NewLineCharacter))
 	}
 	if r.isCompact {
 		r.out.Write([]byte(NewLineCharacter))
-		if !r.IsIncognito {
+		if !r.IsIncognito && assetUrl != "" {
+			// we do not have a space url, so we extract it form the asset url
+			// https://console.mondoo.com/space/fleet/2JtqGyVTZULTW0uwQ5YxXW4nh6Y?spaceId=dazzling-golick-767384
+			// an individual asset url wouldn't make sense here
+			spaceUrlRegexp := regexp.MustCompile(`^(http.*)/[a-zA-Z0-9-]+(\?.+)$`)
+			m := spaceUrlRegexp.FindStringSubmatch(assetUrl)
+			spaceUrl := m[1] + m[2]
+
 			r.out.Write([]byte("Detailed information is already available via the web UI: "))
 			r.out.Write([]byte(spaceUrl + NewLineCharacter))
 		}

